frontend/src/app/features/invite-user-modal/principal/principal.component.ts

```diff
@@ -89,38 +116,54 @@ export class PrincipalComponent implements OnInit {
     schema: null,
   };
 
-  get principalControl() {
+  get messageControl():AbstractControl|null {
+    return this.principalForm.get('message');
+  }
+
+  get roleControl():AbstractControl|null {
+    return this.principalForm.get('role');
+  }
+
+  get principalControl():AbstractControl|null {
     return this.principalForm.get('principal');
   }
 
   get principal():PrincipalLike|undefined {
-    return this.principalControl?.value;
+    return this.principalControl?.value as PrincipalLike|undefined;
+  }
+
+  get role():RoleResource|undefined {
+    return this.roleControl?.value as RoleResource|undefined;
+  }
+
+  get message():string|undefined {
+    return this.messageControl?.value as string|undefined;
   }
 
-  get dynamicFieldsControl() {
+  get dynamicFieldsControl():AbstractControl|null {
     return this.principalForm.get('userDynamicFields');
   }
 
   get customFields():{ [key:string]:any } {
-    return this.dynamicFieldsControl?.value;
+    return this.dynamicFieldsControl?.value as { [key:string]:any };
```

Comment:
Just a minor hint: if you revert this change, line 148 will still has a linter error, but your PR would be green, as you didn't changed this line.